### ISSUE TITLE
[17/10/2022]:  TASK 5 - Deploy Preview ( CI /CD ) Discussion Thread

### ISSUE THREAD

**Issue by okerekechinweotito** [Open-Science-Community-Saudi-Arabia/OSCSA-en-blog#26]
Previous issue opened for TASK 5 is at #8 
Previous PR opened for TASK 5 is at #11 

I made the above issue and corresponding PR some days ago to fix the Netlify deploy preview. Unfortunately the changes I made there did not fix the issue.  However , I have made a stunning discovery

After days of careful research I discovered that the PR deploy-preview is not working simply because the Quarto Publish does not yet support PR deploy-previews . The deploy-preview will not work with Quarto as the feature is still in development and has not yet been integrated into Quarto Publish.  Here is a link to an OPEN issue on Quarto's Repo to support my claims.  [Support netlify PR previews in publish action](https://github.com/quarto-dev/quarto-actions/issues/33)

The above issue was opened by Quarto to create a feature that would address the exact problem we are trying to fix (Netlify Deploy Preview)

In conclusion , this issue cannot be fixed at the moment until Quarto adds support for it as stated [here](https://github.com/quarto-dev/quarto-actions/issues/33)

@alswajiab @BatoolMM

**Comment by okerekechinweotito** [comment]
@alswajiab @BatoolMM  
Please I need feedback on the above

**Comment by BatoolMM** [comment]
Thank you for doing all this research, can you suggest other solutions to preview the page?

**Comment by jesutobi** [comment]
@okerekechinweotito can i join you on this issue, where are you at the moment

**Comment by okerekechinweotito** [comment]
@BatoolMM  I will look into it

**Comment by okerekechinweotito** [comment]
@BatoolMM  @alswajiab 
![Screenshot (57)](https://user-images.githubusercontent.com/65835404/196626524-f27e34dc-afc0-48cc-a0d9-9425bcb45f2d.png)

Inline with the proposed solutions on the image

- (1) I could try using quarto netlify plug-in as described in the  [docs](https://quarto.org/docs/publishing/netlify.html#plugin-configuration)  ( I cannot guarantee it will work)
- (2) There is another option for any PR author to first run Quarto render locally ( Build App into _site folder) and then deploy with the generated output. ( might make messy diffs but it works , you would be able to see the changes on deploy-preview).  
- (3) I believe the option to use the [quarto render action  ](https://github.com/marketplace/actions/quarto-render) and an existing Netlify deploy action will not work because the only quarto render action in the Github marketplace is this one contained in this [repo](https://github.com/marketplace/actions/quarto-render) . It is outdated as stated in this [issue](https://github.com/pommevilla/quarto-render/issues/19) opened on the repo.

Should we go ahead with option 2. Or should I first try option 1

Note : Please I would welcome any corrections on the statements above. None of it is set in stone , as I might be wrong in my conclusions. Please feel free to point me in the right direction if I am wrong. I am here to learn : )  Thank you

**Comment by BatoolMM** [comment]
I want to point out that this uses Quarto https://github.com/ropensci/dev_guide/blob/main/.github/workflows/pr.yml which was pointed out by Maëlle Salmon. Can you have a look at it @okerekechinweotito, thanks again!

**Comment by okerekechinweotito** [comment]
@BatoolMM  The Deploy Preview is now working. I just tested it with a new PR here #34  .  Kindly view the [deploy-preview](https://deploy-preview-34--oscsa-en-blog.netlify.app/) on that PR to confirm. I changed all the blog titles on the test PR to the sentence ["I made this change in the Test PR , the deploy preview is now working"](https://deploy-preview-34--oscsa-en-blog.netlify.app/posts/Internship%20wrap-up/) for demonstration purposes
